src/engine/effects/engineeffectchain.cpp

```diff
@@ -308,12 +312,19 @@ bool EngineEffectChain::process(const ChannelHandle& inputHandle,
                     }
 
                     processingOccured = true;
+                    effectChainGroupDelayFrames += pEffect->getGroupDelayFrames();
+
                     // Output of this effect becomes the input of the next effect
                     pIntermediateInput = pIntermediateOutput;
                 }
             }
         }
 
+        m_pEffectsDelay->setDelayFrames(effectChainGroupDelayFrames);
+        m_pEffectsDelay->process(pIn, m_tmpBuffer.data(), numSamples);
+
+        SampleUtil::copy(pIn, m_tmpBuffer.data(), numSamples);
```

Comment:
It feels scary that pIn is not const. I think It should be, because messing around with the input buffer feels wrong. 

There is also no need for this copy. We can use m_tmpBuffer below.